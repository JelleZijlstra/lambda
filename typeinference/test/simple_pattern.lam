let f = \p. match p with {
	(3, true) -> print 0
	| (2, false) -> print 1
	| (4, y) -> print (if y then 2 else 3)
	| (x, _) -> print x
} in
let f' = \n. match n with {
	0 -> print 1
	| 1 -> print 0
	| _ -> print n
} in
(f (3, true); f (2, false); f (4, true); f' 0; f' 1; f' 2)
