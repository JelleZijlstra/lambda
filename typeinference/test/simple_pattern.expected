%%stdout%%
0
1
2
1
0
2
Result: module
	let __result__ = ()
end
%%stderr%%
